scripts/ts_error_checker.js

```diff
@@ -0,0 +1,199 @@
+const TSCONFIG_PATH = path.join(process.cwd(), 'tsconfig.check.json');
+
+console.log(
+  `${colors.cyan}━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━${colors.reset}`
+);
+console.log(`${colors.bold}TypeScript Error Checker${colors.reset}`);
+console.log(
+  `${colors.cyan}━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━${colors.reset}`
+);
+console.log(
+  `\n${colors.cyan}▶ ${colors.reset}Running TypeScript compiler to check for new errors...\n`
+);
+
+// Load baseline errors
+function loadBaseline() {
+  try {
+    if (fs.existsSync(BASELINE_FILE)) {
+      const content = fs.readFileSync(BASELINE_FILE, 'utf8');
+      const baseline = JSON.parse(content);
+      console.log(`${colors.blue}ℹ ${colors.reset}Loaded baseline from ${BASELINE_FILE}`);
+      console.log(
+        `${colors.blue}ℹ ${colors.reset}Baseline contains ${colors.bold}${baseline.errors.length}${colors.reset} known errors`
+      );
+      return baseline.errors;
+    } else {
+      console.warn(`${colors.yellow}⚠ ${colors.reset}Baseline file not found at ${BASELINE_FILE}.`);
+      console.warn(`${colors.yellow}⚠ ${colors.reset}All errors will be treated as new.`);
+      return [];
+    }
+  } catch (error) {
+    console.error(`${colors.red}✖ ${colors.reset}Error loading baseline: ${error.message}`);
+    process.exit(1);
+  }
+}
+
+// Helper to run TypeScript compiler and get structured output
+function getCurrentErrors() {
+  // Check if tsconfig.json exists
+  if (!fs.existsSync(TSCONFIG_PATH)) {
+    console.error(`Error: Could not find tsconfig.json at ${TSCONFIG_PATH}`);
+    process.exit(1);
+  }
+
+  console.log(`Using TypeScript config: ${TSCONFIG_PATH}`);
+
+  const args = ['--project', TSCONFIG_PATH, '--noEmit', '--pretty', 'false'];
+
+  console.log(`${colors.cyan}▶ ${colors.reset}Executing: ${TSC_BIN} ${args.join(' ')}\n`);
+
+  const result = spawnSync(TSC_BIN, args, {
+    encoding: 'utf8',
+    shell: true,
+    stdio: 'pipe',
+  });
+
+  // Process doesn't throw an exception when there are type errors
+  // Instead we need to check the exit code
+  if (result.status === 0) {
+    // Success - no errors
+    console.log(
+      `${colors.green}✓ ${colors.reset}${colors.bold}No TypeScript errors found!${colors.reset}`
+    );
+    return [];
+  } else {
+    // Parse the error output to get structured error info
+    const errors = [];
+    const output = result.stderr || result.stdout || '';
+
+    output.split('\n').forEach((line) => {
+      const match = line.match(/(.+)\((\d+),(\d+)\):\s+error\s+(TS\d+):\s+(.+)/);
+      if (match) {
+        errors.push({
+          file: match[1],
+          line: parseInt(match[2]),
+          column: parseInt(match[3]),
+          code: match[4],
+          message: match[5],
+        });
+      }
+    });
+
+    console.log(
+      `${colors.blue}ℹ ${colors.reset}Found ${colors.bold}${errors.length}${colors.reset} TypeScript errors in current codebase`
+    );
+    return errors;
```

Comment:
Do we need to sort this error array before returning the result? Will the output order be stable between different machines?